sdk/src/transaction/mod.rs

```diff
@@ -508,7 +508,9 @@ pub fn uses_durable_nonce(tx: &Transaction) -> Option<&CompiledInstruction> {
                 Some(program_id) => system_program::check_id(program_id),
                 _ => false,
             }
-        } && matches!(limited_deserialize(&maybe_ix.data), Ok(SystemInstruction::AdvanceNonceAccount))
+        }
+        && matches!(maybe_ix.accounts.get(0), Some(nonce_idx) if message.is_writable(*nonce_idx as usize, false))
```

Comment:
I'm having a hard time understanding this syntax, do you mind rewriting it?

Reply:
heh, funny, that syntax was compiler suggested after it didn't like my `match`.  You aren't the only one who doesn't like it, `cargo fmt` also just skips over it in disdain ;-)

I'll rewrite it

Reply:
how about now?

Reply:
Looks good, definitely better! I still need to get used to that `matches!` macro though. It hurts my brain to have the `if` in the middle